### ISSUE TITLE
About Loss

### ISSUE THREAD

**Issue by skr998** [wvangansbeke/Unsupervised-Classification#100]
Hello, thanks for your great work.

The loss function you define in your paper:
![image](https://user-images.githubusercontent.com/77918488/147875145-061bcc3a-0ad4-4473-a379-8fe289c68339.png)
There is a question about it. Obviously, the first term is always positive, but the second term is always negative. So the two items add up to cancel each other out. I want to know why not set them both positive or negative?

**Comment by wvangansbeke** [comment]
Hi @skr998,

No it makes sense that the signs are opposite: The first term maximizes the similarity with its K nearest neighbors by minimizing the dot-product (pull-force). The second term maximizes the entropy over the class assignments (push-force).
Please checkout the paper for more details on the loss function.

**Comment by Dadatata-JZ** [comment]
@wvangansbeke hi Wounter, when I ran the pipeline with my customized data, at scan step, I received negative loss. I assume it may be because `total_loss = - entropy_loss + consistency_loss`, where entropy is greater than consistency loss. Do you have any suggestions?

**Comment by Pro-xiaowen** [comment]
@Dadatata-JZ  Have you solved this problem about negative loss  using your own data? I also encountered this problem

**Comment by Pro-xiaowen** [comment]
> @Dadatata-JZ Have you solved this problem about negative loss using your own data? I also encountered this problem

@wvangansbeke  Colud you can give some suggestions?
